ecs-web/main.tf

```diff
@@ -46,70 +47,78 @@ resource "aws_ecs_task_definition" "concourse_web_task_definition" {
-    concourse_db_host              = "${var.concourse_db_host}"
-    concourse_db_port              = "${var.concourse_db_port}"
-    concourse_db_user              = "${var.concourse_db_username}"
-    concourse_db_password          = "${jsonencode(var.concourse_db_password)}"
-    concourse_db_name              = "${var.concourse_db_name}"
-    awslog_group_name              = "${aws_cloudwatch_log_group.concourse_web_log_group.name}"
-    awslog_region                  = "${data.aws_region.current.name}"
-    concourse_keys_bucket_name     = "${var.keys_bucket_id}"
-    concourse_basic_auth           = "${length(var.concourse_auth_username) > 0 && length(var.concourse_auth_password) > 0 ? data.template_file.concourse_basic_auth.rendered : ""}"
-    concourse_basic_auth_main_team = "${length(var.concourse_auth_main_team_local_user) > 0 ? data.template_file.concourse_basic_auth_main_team_local_user.rendered : ""}"
-    concourse_github_auth          = "${length(var.concourse_github_auth_client_id) > 0 && length(var.concourse_github_auth_client_secret) > 0 && length(var.concourse_github_auth_team) > 0 ? data.template_file.concourse_github_auth.rendered : ""}"
-    concourse_vault_variables      = "${length(var.vault_server_url) > 0 ? data.template_file.concourse_vault_variables.rendered : ""}"
-    memory                         = "${var.container_memory}"
-    cpu                            = "${var.container_cpu}"
-    concourse_prometheus_bind_port = "${var.concourse_prometheus_bind_port}"
-    concourse_prometheus_bind_ip   = "${var.concourse_prometheus_bind_ip}"
-    concourse_db_task_definition   = "${indent(2, join("", data.template_file.concourse_db_task_template.*.rendered))}"
-    volumes_from_concourse_db      = "${var.auto_create_db ? ",{ \"sourceContainer\": \"create_db\" }" : ""}"
-    volumes_from_vault_auth        = "${length(var.vault_server_url) > 0 ? ",{ \"sourceContainer\": \"vault_auth\" }" : ""}"
-    vault_command_args             = "${length(var.vault_server_url) > 0 ? "--vault-client-token=`cat /concourse_vault/token`" : ""}"
-    vault_auth_task_definition     = "${indent(2, join("", data.template_file.vault_auth_task_template.*.rendered))}"
+    concourse_hostname             = local.concourse_hostname
+    concourse_db_host              = var.concourse_db_host
+    concourse_db_port              = var.concourse_db_port
+    concourse_db_user              = var.concourse_db_username
+    concourse_db_password          = jsonencode(var.concourse_db_password)
+    concourse_db_name              = var.concourse_db_name
+    awslog_group_name              = aws_cloudwatch_log_group.concourse_web_log_group.name
+    awslog_region                  = data.aws_region.current.name
+    concourse_keys_bucket_name     = var.keys_bucket_id
+    concourse_basic_auth           = length(var.concourse_auth_username) > 0 && length(var.concourse_auth_password) > 0 ? data.template_file.concourse_basic_auth.rendered : ""
```

Comment:
Use `null` as default for:
- `concourse_auth_username`
- `concourse_auth_password`
- `concourse_auth_main_team_local_user`
- `concourse_github_auth_client_id`
- `concourse_github_auth_client_secret`
- `concourse_github_auth_team`
- `vault_server_url`
- `vault_auth_concourse_role_name`
- `concourse_db_postgres_engine_version`

Reply:
For data templates with null values I had to do something like the following, 
``` 
 vars = {
    concourse_vault_url = coalesce(var.vault_server_url, 0)
  }
```
because Terraform renders the template even if condition is not met, I've also tested that with `locals` and got the same behavior. 
I know It's not a very decent workaround but after all the `0` value is not used anywhere. If you have any other ideas let me know :)